### PR TITLE
Check if the repository for the transpile task is valid before running the evaluation, so it is checked just once

### PR DESCRIPTION
**Pull request by ruiAzevedo19**
Part of #263

### CODE REVIEW

File: evaluate/task/task-transpile.go [hunk 1]
```diff
@@ -191,3 +187,44 @@ func (t *TaskTranspile) unpackTranspilerPackage(ctx evaltask.Context, fileLogger
...
+// validateTranspileRepository checks if the repository for the "transpile" task is well-formed.
+func validateTranspileRepository(logger *log.Logger, repositoryPath string, destinationLanguage language.Language) (err error) {
+	logger.Printf("validating repository %q", repositoryPath)
+
+	var originLanguage language.Language
+	if _, ok := destinationLanguage.(*golang.Language); ok {
+		originLanguage = &java.Language{}
+	} else {
+		originLanguage = &golang.Language{}
+	}
```
Comment (bauersimon) [inline evaluate/task/task-transpile.go:200]: Can you please already open the Ruby support issue and note that these switches need to be adapted (we do this somewhere else as well AFAIK)?

**Review by bauersimon — APPROVED** [review]

File: evaluate/task/task-transpile.go (hunk 1 above)
Comment (bauersimon) [reply evaluate/task/task-transpile.go:200]: Ah the ruby support issue exists already. Will put it there.

Comment (ruiAzevedo19) [reply evaluate/task/task-transpile.go:200]: We also do it in the `Run` method.

Comment (ruiAzevedo19) [reply evaluate/task/task-transpile.go:200]: Added to https://github.com/symflower/eval-dev-quality/issues/300

**Pull request merged**